WIP: issue-116-emily #116: Guideline against "pseudo modules" is problematic (#161)
redhat-documentation/modular-docs: modular-docs-manual/content/topics/using_text_snippets_or_text_fragments.adoc

[id="using_text_snippets_or_text_fragments_{context}"]

> = Text Snippets or Text Fragments

Reviewer: ```suggestion
= Text snippets or text fragments
```

Author: Thanks @ncbaratta . I make this title case on purpose to be parallel with the other headings in the guide, for now. We have an issue to fix all of these.

Reviewer 2: Let's see if my comment triggers any other bites.  Maybe consider removing "text snippet" because Atom has a feature that uses the term "snippet". (I don't use Atom, so I can't elaborate.)  "Text fragment" throughout this section ought to be sufficiently clear.

Author: I agree, Mike. I removed 'snippet'.

> A text snippet or text fragment is a small section of text that is stored in an Asciidoc file. Text snippets contain context that is resused, for example:

Reviewer: ```suggestion
A text snippet or text fragment is a small section of text that is stored in an Asciidoc file. Text snippets contain content that is reused in multiple modules, for example:
```

* A step or series of steps in a procedure

> * A disclaimer, for example for technology preview or beta releases

Reviewer: ```suggestion
* A disclaimer, for example, for technology preview or beta releases
```

> NOTE: A text snippet is not a module. The Asciidoc file that contains the text snippet cannot contain any other information. A module cannot contain and include statement for another module.

Reviewer: ```suggestion
NOTE: A text snippet is not a module. The Asciidoc file that contains the text snippet cannot contain any other information. A module cannot contain an include statement for another module.
```

> NOTE: A text snippet is not a module. The Asciidoc file that contains the text snippet cannot contain any other information. A module cannot contain an include statement for another module.

Reviewer 2: IMO, not needed.  The examples in the bullets are clear.

Author: @mikemckiernan I would like to leave this note. In the past, we have had issues with writers including modules in modules and we want to make sure to communicate that doing this is not OK. Also, we want to make sure that the asciidoc file is 'clean'.

Reviewer 2: np from me.

.Procedure

> . Create the text snippet Asciidoc file and store it in your snippets folder.

Reviewer 2: I'll lick a fingertip and then reach out for the third rail...vanilla Asciidoctor is perfectly happy with a snippets folder.  Does the tooling that we're not supposed to be overly focused on also support a snippets folder?

Other than mental harm, is it problematic to put the text frags in a modules folder?--I get it that we just stated it's not a module, then we're possibly advising to co-locate with the modules.

Author: Yes, the Publishing-Platform-That-Shall-Not-Be-Named needs to have fragments in a single location. This location is added to the YAML file.

Reviewer 2: I'd swear that the file now includes mention of a "fragments" folder, but I did not checkout the PR.  I share that in case it was an overzealous change from "snippets" to "fragments." I'm good with whatever you propose.
